scripts/generator/graphql_generator/csharp/SDK/iOS/iOSWebCheckout.cs.erb

```diff
@@ -0,0 +1,42 @@
+namespace <%= namespace %>.SDK.iOS {
+#if !SHOPIFY_MONO_UNIT_TEST
+    using UnityEngine;
+    using System.Runtime.InteropServices;
+    using <%= namespace %>.SDK;
+
+    class iOSWebCheckout: IWebCheckout {
+        private static iOSWebCheckoutBehaviour webComponent;
+        private Cart CurrentCart;
+
+        [DllImport ("__Internal")]
+        private static extern void _StartWebCheckout(string unityDelegateObjectName, string checkoutURL);
+
+        public iOSWebCheckout(Cart cart) {
+            CurrentCart = cart;
+        }
+
+        public void Start(string checkoutURL) {
+            if (webComponent == null) {
+                webComponent = GlobalGameObject.AddComponent<iOSWebCheckoutBehaviour>();
+            }
+            webComponent.CurrentCart = CurrentCart;
+            _StartWebCheckout(GlobalGameObject.GameObjectName, checkoutURL);
+        }
+    }
+
+    internal class iOSWebCheckoutBehaviour: MonoBehaviour {
+        internal Cart CurrentCart; 
+
+        internal void OnNativeMessage(string jsonMessage) {
+            var message = NativeMessage.CreateFromJSON(jsonMessage);
+            if (message.Content == "loadedThankYou") {
+                // Do a round trip to the server to see if the purchase is solid.
+                // Invoke callbacks on cart to pass to Unity
+                CurrentCart.onCheckoutSucceeded.Invoke();
```

Comment:
Yea I agree - it would be cleaner to do so. We could pass in delegates into iOSWebCheckout prior to checking out like so:

```csharp
// (in Cart.cs)
void StartWebCheckout() {
    webCheckout.OnCheckoutSucceeded = () => {
        // Call external calback
    };
    webCheckout.OnCheckoutFailed = () => {
        ....
    };
    webCheckout.Start(url);
}
```